src/Umbraco.Web.UI.NetCore/Umbraco/PartialViewMacros/Templates/Breadcrumb.cshtml

```diff
@@ -18,7 +18,7 @@
         @* For each page in the ancestors collection which have been ordered by Level (so we start with the highest top node first) *@
         @foreach (var item in selection.OrderBy(x => x.Level))
         {
-            <li><a href="@item.Url()">@item.Name</a> <span class="divider">/</span></li>
+            <li><a href="@item.Url(PublishedUrlProvider)">@item.Name</a> <span class="divider">/</span></li>
```

Comment:
This will be an interesting change for people if we make this a requirement. I'm all for it since the model shouldn't be responsible for generating the URL but I think if we make the change we shouldn't use ext methods on the model to do it and instead just call the service. But we can discuss later.

Reply:
I guess we will end up having an extra nuget package. with the old codesmells. So having the old extention methods available. but we should not use these our self, or in templates.. Code is untestable using these..

`Umbraco.Friendly` or similar sounds good ;)
At least better than `Umbraco.ReintroduceCodeSmells` or `Umbraco.ReintroduceUntestableCode` :P

Reply:
Yep we'll see how it goes, i think the more we move away from the hacks the better and then can re-implement 'friendly' things later if necessary. In some cases though i think we'll realize that some of the 'friendly' things don't actually make much sense. For example, when (one day) we decouple the content cache from URLs, then item.Url() probably won't make sense anymore.